hypervisor/scripts/gen_acrn_deb.py

```diff
@@ -0,0 +1,175 @@
+        source = cur_dir + '/../' + source
+        target = deb_dir + target
+
+        if os.path.exists(target):
+            run_command('cp %s %s' % (source, target), cur_dir)
+        else:
+            run_command('mkdir -p %s' % target, cur_dir)
+            run_command('cp %s %s' % (source, target), cur_dir)
+
+    run_command('cp ./misc/packaging/acrn-hypervisor.postinst ./build/acrn_release_deb/DEBIAN/postinst', cur_dir)
+    run_command('chmod +x ./build/acrn_release_deb/etc/grub.d/100_ACRN', cur_dir)
+    run_command('chmod +x ./build/acrn_release_deb/DEBIAN/postinst', cur_dir)
+    run_command('sed -i \'s/\r//\' ./build/acrn_release_deb/DEBIAN/postinst', cur_dir)
+
+    run_command('cp ./misc/packaging/acrn-hypervisor.preinst ./build/acrn_release_deb/DEBIAN/preinst', cur_dir)
+    run_command('chmod +x ./build/acrn_release_deb/DEBIAN/preinst', cur_dir)
+    run_command('sed -i \'s/\r//\' ./build/acrn_release_deb/DEBIAN/preinst', cur_dir)
+
+    run_command('dpkg -b acrn_release_deb acrn-%s-%s-%s.deb' %(board, scenario, version), build_dir)
+
+    return
+
+def create_acrn_board_inspector_deb(cur_dir, deb_dir, version, build_dir):
+    #control file description
+    listcontrol=['Package: acrn-board-inspector\n',
+                'version: %s \n'% version,
+                'Section: free \n',
+                'Priority: optional \n',
+                'Architecture: amd64 \n',
+                'Maintainer: Intel\n',
+                'Description: ACRN board inspector tools \n',
+                'Depends: cpuid, msr-tools, pciutils, dmidecode, python3, python3-pip, python3-lxml \n',
+                '\n']
+    with open(deb_dir + '/DEBIAN/control','w',encoding='utf-8') as fr:
+            fr.writelines(listcontrol)
+    run_command('cp -r ./misc/config_tools/board_inspector/ ./build/acrn_board_inspector_deb/bin/', cur_dir)
+    run_command('cp ./misc/packaging/acrn-board-inspector.postinst ./build/acrn_board_inspector_deb/DEBIAN/postinst', cur_dir)
+    run_command('chmod +x ./build/acrn_board_inspector_deb/DEBIAN/postinst', cur_dir)
+    run_command('sed -i \'s/\r//\' ./build/acrn_board_inspector_deb/DEBIAN/postinst', cur_dir)
+    run_command('cp ./misc/packaging/acrn-board-inspector.prerm ./build/acrn_board_inspector_deb/DEBIAN/prerm', cur_dir)
+    run_command('chmod +x ./build/acrn_board_inspector_deb/DEBIAN/prerm', cur_dir)
+    run_command('sed -i \'s/\r//\' ./build/acrn_board_inspector_deb/DEBIAN/prerm', cur_dir)
+
+    run_command('dpkg -b acrn_board_inspector_deb acrn-board-inspector-%s.deb' %(version), build_dir)
+
+    return
+
+if __name__ == "__main__":
+    if len(sys.argv) < 5:
+        BUILD_DIR = sys.argv[1]
+        HV_VERSION = sys.argv[2]
+        DEB_MODE = sys.argv[3]
+    else:
+        BOARD = sys.argv[1]
+        SCENARIO = sys.argv[2]
+        BUILD_DIR = sys.argv[3]
+        HV_VERSION = sys.argv[4]
+        DEB_MODE = 'no'
+
+    cur_dir = os.getcwd()
+    if DEB_MODE != 'board_inspect':
```

Comment:
It seems Python is not a good choice for this script as most of your code are invoking external tools via the shell. Why not using a shell script instead?

Reply:
Will this be rewritten as a shell script as @junjiemao1 recommends?